### DISCUSSION TITLE
Switch expression returns null or the default value (0) for nullable Enum.

### DISCUSSION THREAD

**Discussion by djs2014** [dotnet/csharplang#3958] [Q&A] [answered]
# Switch expression returns a null or Enum default value for nullable Enum.

OS: windows 10 and Linux Mint 20 Cinnamon - kernel 5.4.0-48-generic
SDK: .NET Core SDK (3.1.402)

# General

In example 1 the value of orientation is Orientation.North
``` 
// example 1
            var direction = Directions.Up;
            Console.Write($"Map view direction is {direction}");

            Orientation? orientation = direction switch
            {
                Directions.Up => default,
                Directions.Right => Orientation.East,
                Directions.Down => Orientation.South,
                Directions.Left => Orientation.West,
            };
```
In example 2 the value of orientation is null
```
// example 2
            var direction = Directions.Right;
            Console.Write($"Map view direction is {direction}");

            Orientation? orientation = direction switch
            {
                Directions.Right => default,
            };
```

```
public enum Directions
    {
        Up,
        Down,
        Right,
        Left
    }

    public enum Orientation
    {
        North,
        South,
        East,
        West
    }
```
Unit test:
[TestSwitch.zip](https://github.com/dotnet/core/files/5281442/TestSwitch.zip)

**Comment by svick** [comment, marked as answer]
I believe this is the expected behavior. The `switch` expression is target-typed, but only when there isn't a best common type for the right-hand side expressions. 

The best common type of `Orientation.East` and `default` is `Orientation`, so `default` becomes `default(Orientation)`.

The best common type of `default` doesn't exist, so target-typing is used and `default` becomes `default(Orientation?)`.

In these cases, where it's not clear what `default` means, I'd suggest being more explicit and writing something like `Orientation.North` or `0`, or `null`.

**Reply by 333fred** [reply]
Yes, this expected behavior for a switch expression. I would also agree that using `default` in a context where it could be either a value type or a reference type is confusing, and you should be explicit.

**Reply by djs2014** [reply]
Thank you for the answer, the clearest way will then be:
```
// example 1
            var direction = Directions.Up;
            Console.Write($"Map view direction is {direction}");

            Orientation? orientation = direction switch
            {
                Directions.Up => default(Orientation?),
                Directions.Right => Orientation.East,
                Directions.Down => Orientation.South,
                Directions.Left => Orientation.West,
            };
```
`orientation` will then be `null`.

**Reply by jnm2** [reply]
I think `null` is clearer than `default` in every case.

**Comment by danmoseley** [comment]
@jaredpar can you pleae either close, or move to csharp-lang? I don't have rights to do so.

**Comment by alrz** [comment]
https://github.com/dotnet/roslyn/issues/38270
